src/Sfa.Roatp.Indexer.ApplicationServices/RoatpProvidersXlsxService.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using OfficeOpenXml;
+using Sfa.Roatp.Indexer.ApplicationServices.Settings;
+using Sfa.Roatp.Indexer.Core.Models;
+
+namespace Sfa.Roatp.Indexer.ApplicationServices
+{
+    public class RoatpProvidersXlsxService : IGetRoatpProviders
+    {
+        private const int UkprnPosition = 1;
+        private const int ProviderTypePosition = 3;
+        private const int ContractedForNonLeviedEmployersPosition = 4;
+        private const int ParentCompanyGuaranteePosition = 5;
+        private const int NewOrganisationWithoutFinancialTrackRecordPosition = 6;
+        private const int StartDatePosition = 7;
+        private const int EndDatePosition = 8;
+
+        private readonly IAppServiceSettings _appServiceSettings;
+
+        public RoatpProvidersXlsxService(IAppServiceSettings appServiceSettings)
+        {
+            _appServiceSettings = appServiceSettings;
+        }
+
+        public List<RoatpProvider> GetRoatpData()
+        {
+            var roatpProviders = new List<RoatpProvider>();
+
+            using (var client = new WebClient())
+            {
+                if (!string.IsNullOrEmpty(_appServiceSettings.GitUsername))
+                {
+                    var credentials = Convert.ToBase64String(Encoding.ASCII.GetBytes($"{_appServiceSettings.GitUsername}:{_appServiceSettings.GitPassword}"));
+                    client.Headers[HttpRequestHeader.Authorization] = $"Basic {credentials}";
+                }
+
+                var filePath = Path.GetTempFileName();
+                client.DownloadFile(new Uri(_appServiceSettings.VstsRoatpUrl), filePath);
+
+                using (ExcelPackage package = new ExcelPackage(new FileInfo(filePath)))
+                {
+                    GetRoatpProviders(package, roatpProviders);
+                }
+            }
+
+            var response = roatpProviders.Where(roatpProviderResult => roatpProviderResult.Ukprn != string.Empty).ToList();
```

Comment:
Could remove .ToList and leave this as IEnumerable and in the interface as well.